gobblin-utility/src/main/java/gobblin/util/AvroFlattener.java

```diff
@@ -0,0 +1,357 @@
+
+    LOG.debug("Original Schema : " + schema);
+    LOG.debug("Flattened Schema: " + flattenedSchema);
+
+    return flattenedSchema;
+  }
+
+  private static Schema flatten(Schema schema, boolean shouldPopulateLineage) {
+    Schema flattenedSchema;
+
+    // Process all Schema Types
+    // (Primitives are simply cloned)
+    switch (schema.getType()) {
+      case ARRAY:
+        // Array might be an array of recursive Records, flatten them
+        flattenedSchema = Schema.createArray(flatten(schema.getElementType(), false));
+        break;
+      case BOOLEAN:
+        flattenedSchema = Schema.create(schema.getType());
+        break;
+      case BYTES:
+        flattenedSchema = Schema.create(schema.getType());
+        break;
+      case DOUBLE:
+        flattenedSchema = Schema.create(schema.getType());
+        break;
+      case ENUM:
+        flattenedSchema =
+            Schema.createEnum(schema.getName(), schema.getDoc(), schema.getNamespace(), schema.getEnumSymbols());
+        break;
+      case FIXED:
+        flattenedSchema =
+            Schema.createFixed(schema.getName(), schema.getDoc(), schema.getNamespace(), schema.getFixedSize());
+        break;
+      case FLOAT:
+        flattenedSchema = Schema.create(schema.getType());
+        break;
+      case INT:
+        flattenedSchema = Schema.create(schema.getType());
+        break;
+      case LONG:
+        flattenedSchema = Schema.create(schema.getType());
+        break;
+      case MAP:
+        flattenedSchema = Schema.createMap(flatten(schema.getValueType(), false));
+        break;
+      case NULL:
+        flattenedSchema = Schema.create(schema.getType());
+        break;
+      case RECORD:
+        flattenedSchema = flattenRecord(schema, shouldPopulateLineage);
+        break;
+      case STRING:
+        flattenedSchema = Schema.create(schema.getType());
+        break;
+      case UNION:
+        flattenedSchema = flattenUnion(schema, shouldPopulateLineage);
+        break;
+      default:
+        String exceptionMessage = String.format("Schema flattening failed for \"%s\" ", schema);
+        LOG.error(exceptionMessage);
+
+        throw new AvroRuntimeException(exceptionMessage);
+    }
+
+    // Copy schema metadata
+    copyProperties(schema, flattenedSchema);
+
+    return flattenedSchema;
+  }
+
+  /***
+   * Flatten Record schema
+   * @param schema Record Schema to flatten
+   * @param shouldPopulateLineage If lineage information should be tagged in the field, this is true when we are
+   *                              un-nesting fields
+   * @return Flattened Record Schema
+   */
+  private static Schema flattenRecord(Schema schema, boolean shouldPopulateLineage) {
+    Preconditions.checkNotNull(schema);
+    Preconditions.checkArgument(Schema.Type.RECORD.equals(schema.getType()));
+
+    Schema flattenedSchema;
+
+    List<Schema.Field> flattenedFields = new ArrayList<>();
+    if (null != schema.getFields() && schema.getFields().size() > 0) {
```

Comment:
This line is unnecessary since `schema.getFields()` won't return null. Same for line 132.